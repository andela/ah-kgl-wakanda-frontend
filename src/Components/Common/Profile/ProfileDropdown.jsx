import React from 'react';
import PropTypes from 'prop-types';
import 'bootstrap/dist/css/bootstrap.css';
import '@fortawesome/fontawesome-free/css/all.min.css';
import { Dropdown, Image } from 'react-bootstrap';
import './ProfileDropdown.scss';
import image from '../../../assets/img/images.png';
import DropdownItem from './DropdownItem';

/**
 * @param {*} props
 * @returns {void}
 */
const ProfileDropdown = props => {
  const { firstName, lastName, picture } = props;
  return (
    <React.Fragment>
      <Dropdown>
        <Dropdown.Toggle variant="light" id="dropdown-basic">
          <Image src={picture} roundedCircle />
          <span className="names">
            {firstName}
            {lastName}
          </span>
        </Dropdown.Toggle>
        <DropdownItem />
      </Dropdown>
    </React.Fragment>
  );
};

ProfileDropdown.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  picture: PropTypes.string,
};
ProfileDropdown.defaultProps = {
  picture: { image },
};

export default ProfileDropdown;
