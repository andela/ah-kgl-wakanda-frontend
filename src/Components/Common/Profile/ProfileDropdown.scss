#dropdown-basic {
  place-items: 10px;
  background-color: transparent;
  box-shadow: none;
  border: none;
  img {
    width: 30px;
  }
  #rounded-circle {
    width: 30px;
  }
  .names {
    padding: 0 8px;
    font-size: 0.9em;
  }
}
