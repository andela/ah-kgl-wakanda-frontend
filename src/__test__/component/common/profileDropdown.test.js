import React from 'react';
import { shallow } from 'enzyme';
import ProfileDropdown from '../../../Components/Common/Profile/ProfileDropdown';
import img from '../../../assets/img/images.png';

const props = {
  username: '@gisel',
  picture: img,
};
describe('Render the Profile Dropdown component', () => {
  it('should display button', async () => {
    const dropdown = shallow(<ProfileDropdown {...props} />);
    expect(dropdown).toMatchSnapshot();
    expect(dropdown.find('button'));
    expect(dropdown.find('img'));
    expect(dropdown.find('span'));
  });
  it('should display the dropdown', async () => {
    const dropdown = shallow(<ProfileDropdown {...props} />);
    expect(dropdown.find('.dropdown-menu'));
    expect(dropdown.find('a'));
  });
});
