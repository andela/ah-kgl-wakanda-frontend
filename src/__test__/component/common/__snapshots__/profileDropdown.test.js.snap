// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Profile Dropdown component should display button 1`] = `ShallowWrapper {}`;
